frontend/scripts/pipelinespec_pbjs.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 The Kubeflow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Note: This script uses protobuf.js to generate pipeline_spec classes in Typescript.
+// Reason: protoc doesn't decode from plain json payload to proto object.
+//         protobuf.js supports this feature along with other protobuf functionality.
+// Usage: Under frontend/, run: `npm run build:pipeline-spec`.
+// Output: Result is stored in frontend/src/third_party/pipeline_spec.
```

Comment:
nit: pipelinespec is not third party, you may put it in a folder called `generated`

Reply:
Sounds good, updated the destination to `generated/pipeline_spec`.